zerver/tests/test_soft_deactivation.py

```diff
@@ -78,6 +78,10 @@ def test_do_soft_activate_users(self) -> None:
             self.example_user('iago'),
             self.example_user('cordelia'),
         ]
+        # Cover the case where a user unsubscribes before
+        # getting soft deactivated.
+        self.unsubscribe(users[0], 'Denmark')
+
```

Comment:
This test is mostly useless coverage.

I think we want this case to be a test that features sending messages to the stream before/after the user is unsubscribed, and that actually checks that the right thing happened in terms of whether the user receives the UserMessage rows for those messages.